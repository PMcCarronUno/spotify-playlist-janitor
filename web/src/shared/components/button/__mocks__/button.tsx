import React from 'react';

export const PrimaryButton = () => <div>Primary Button</div>;
export const SecondaryButton = () => <div>Secondary Button</div>;
