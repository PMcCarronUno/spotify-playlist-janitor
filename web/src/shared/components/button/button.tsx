import { ButtonHTMLAttributes, DetailedHTMLProps } from "react";
import styled from "styled-components";

import { GREEN } from "shared/constants";

type ButtonProps = DetailedHTMLProps<
  ButtonHTMLAttributes<HTMLButtonElement>,
  HTMLButtonElement
>;

const StyledButton = styled.button`
  border-radius: 5rem;
  height: 50px;
  min-width: 112px;
  color: black;
  font-size: 1rem;
  font-weight: 600;
  cursor: pointer;
  border: none;

  &:hover {
    transform: scale(1.04);
  }
`;

const StyledPrimaryButton = styled(StyledButton)`
  background-color: ${GREEN};
`;

const StyledSecondaryButton = styled(StyledButton)`
  background-color: white;
`;

export const PrimaryButton = ({ children, ...rest }: ButtonProps) => {
  return <StyledPrimaryButton {...rest}>{children}</StyledPrimaryButton>;
};

PrimaryButton.displayName = "PrimaryButton";

export const SecondaryButton = ({ children, ...rest }: ButtonProps) => {
  return <StyledSecondaryButton {...rest}>{children}</StyledSecondaryButton>;
};

SecondaryButton.displayName = "SecondaryButton";
