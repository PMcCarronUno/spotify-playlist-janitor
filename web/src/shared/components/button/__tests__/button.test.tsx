import React from "react";
import { fireEvent, getByTestId, render } from "@testing-library/react";

import { PrimaryButton, SecondaryButton } from "../button";

describe("<PrimaryButton />", () => {
  let container: HTMLElement;
  let props: any;
  const mockOnClick = jest.fn();

  beforeEach(() => {
    props = {
      onClick: mockOnClick,
      id: "primary-button-id",
      "data-testid": "test-primary-button-id",
    };

    ({ container } = render(
      <PrimaryButton {...props}>Test Primary Button</PrimaryButton>
    ));
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should render PrimaryButton component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should execute onClick when PrimaryButton clicked", () => {
    const button = getByTestId(container, "test-primary-button-id");
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalled();
  });
});

describe("<SecondaryButton />", () => {
  let container: HTMLElement;
  let props: any;
  const mockOnClick = jest.fn();

  beforeEach(() => {
    props = {
      onClick: mockOnClick,
      id: "secondary-button-id",
      "data-testid": "test-secondary-button-id",
    };

    ({ container } = render(
      <SecondaryButton {...props}>Test Secondary Button</SecondaryButton>
    ));
  });

  it("should render SecondaryButton component", () => {
    expect(container.firstChild).toMatchSnapshot();
  });

  it("should execute onClick when SecondaryButton clicked", () => {
    const button = getByTestId(container, "test-secondary-button-id");
    fireEvent.click(button);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
