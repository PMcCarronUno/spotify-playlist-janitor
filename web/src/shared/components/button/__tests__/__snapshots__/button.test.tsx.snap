// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrimaryButton /> should render PrimaryButton component 1`] = `
<button
  class="sc-beyTiQ sc-guDMob bFOxeO bDhOyw"
  data-testid="test-primary-button-id"
  id="primary-button-id"
>
  Test Primary Button
</button>
`;

exports[`<SecondaryButton /> should render SecondaryButton component 1`] = `
<button
  class="sc-beyTiQ sc-dmyDmy bFOxeO kBSTbV"
  data-testid="test-secondary-button-id"
  id="secondary-button-id"
>
  Test Secondary Button
</button>
`;
